### PR TITLE
feat(chat-e2e): implemented chat selection tests

### PR DESCRIPTION
**Pull request by nartovm**
**Description:**

implemented chat selection tests

**Checklist:**

- [ ] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [ ] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [ ] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by nartovm** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1884.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by nartovm** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1884.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by nartovm** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1884.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by nartovm** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1884.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat-e2e/src/tests/chatSelectionFunctionality.test.ts [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+        { name: firstConversation.name },
+        'visible',
+      );
+    });
+
+    await dialTest.step(
+      'Click on the menu and hover over any item at the bottom (e.g. Delete)',
+      async () => {
+        await conversations.openEntityDropdownMenu(firstConversation.name);
+        await conversations
```
Comment (irinakartun) [inline apps/chat-e2e/src/tests/chatSelectionFunctionality.test.ts:88]: you can use `conversationDropdownMenu` fixture to select menu option

File: apps/chat-e2e/src/tests/chatSelectionFunctionality.test.ts [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+      await conversations
+        .getDropdownMenu()
+        .selectMenuOption(MenuOptions.export);
+      const downloadedData = await dialHomePage.downloadData(() =>
+        conversations
+          .getDropdownMenu()
+          .selectMenuOption(MenuOptions.withoutAttachments),
+      );
+      expect(downloadedData.path).toBeTruthy();
+      expect(downloadedData.path).toMatch(/\.json$/);
```
Comment (irinakartun) [inline apps/chat-e2e/src/tests/chatSelectionFunctionality.test.ts]: we have such verification in other tests. 
The goal of current test is to check that exported conversation is selected. I'd leave only assertion below

File: apps/chat-e2e/src/tests/chatSelectionFunctionality.test.ts [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+          .selectMenuOption(MenuOptions.delete);
+        await confirmationDialog.confirm({ triggeredHttpMethod: 'DELETE' });
+        await conversationAssertion.assertSelectedConversation(
+          playbackConversation,
+        );
+        await conversationAssertion.assertEntityState(
+          { name: replayConversation },
+          'hidden',
+        );
+        replayConversation = '';
```
Comment (irinakartun) [inline apps/chat-e2e/src/tests/chatSelectionFunctionality.test.ts]: why

File: apps/chat-e2e/src/tests/chatSelectionFunctionality.test.ts [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+
+    await dialTest.step('Click on Compare', async () => {
+      await conversations.openEntityDropdownMenu(firstConversation.name);
+      await conversations
+        .getDropdownMenu()
+        .selectMenuOption(MenuOptions.compare);
+      await conversationAssertion.assertSelectedConversation(
+        firstConversation.name,
+      );
+      await expect
```
Comment (irinakartun) [inline apps/chat-e2e/src/tests/chatSelectionFunctionality.test.ts:124]: please, use ` await conversationAssertion.assertEntityState()` instead

**Review by irinakartun — CHANGES_REQUESTED** [review]

File: apps/chat-e2e/src/tests/chatSelectionFunctionality.test.ts [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+      await conversations
+        .getDropdownMenu()
+        .selectMenuOption(MenuOptions.duplicate);
+      clonedConversation = `${firstConversation.name} 1`;
+      await conversationAssertion.assertSelectedConversation(
+        clonedConversation,
+      );
+    });
+
+    await dialTest.step('Click on Share', async () => {
```
Comment (irinakartun) [inline apps/chat-e2e/src/tests/chatSelectionFunctionality.test.ts:224]: this is automated in scope of sharedChatIcons.test.ts tests, can be removed

File: apps/chat-e2e/src/tests/chatSelectionFunctionality.test.ts (hunk 1 above)
Comment (nartovm) [reply apps/chat-e2e/src/tests/chatSelectionFunctionality.test.ts:88]: done

File: apps/chat-e2e/src/tests/chatSelectionFunctionality.test.ts (hunk 2 above)
Comment (nartovm) [reply apps/chat-e2e/src/tests/chatSelectionFunctionality.test.ts]: will not be done

File: apps/chat-e2e/src/tests/chatSelectionFunctionality.test.ts (hunk 3 above)
Comment (nartovm) [reply apps/chat-e2e/src/tests/chatSelectionFunctionality.test.ts]: Removed

File: apps/chat-e2e/src/tests/chatSelectionFunctionality.test.ts (hunk 4 above)
Comment (nartovm) [reply apps/chat-e2e/src/tests/chatSelectionFunctionality.test.ts:124]: It is not necessary

File: apps/chat-e2e/src/tests/chatSelectionFunctionality.test.ts (hunk 5 above)
Comment (nartovm) [reply apps/chat-e2e/src/tests/chatSelectionFunctionality.test.ts:224]: Updated

**Comment by nartovm** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1884.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat-e2e/src/tests/chatSelectionFunctionality.test.ts [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+      clonedConversation = `${firstConversation.name} 1`;
+      await conversations.getEntityByName(clonedConversation).waitFor();
+      await conversationAssertion.assertSelectedConversation(
+        clonedConversation,
+      );
+    });
+
+    await dialTest.step('Click on Share', async () => {
+      await conversations.openEntityDropdownMenu(playbackConversation);
+      await conversations.getDropdownMenu().selectMenuOption(MenuOptions.share);
```
Comment (irinakartun) [inline apps/chat-e2e/src/tests/chatSelectionFunctionality.test.ts]: await conversations.getDropdownMenu().selectMenuOption() -> await conversationDropdownMenu.selectMenuOption()

**Review by irinakartun — DISMISSED** [review]

**Comment by nartovm** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1884.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by taa-epam** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1884.nightly-test.deltixhub.io
>E2E tests status: failed

File: apps/chat-e2e/src/tests/chatSelectionFunctionality.test.ts (hunk 6 above)
Comment (nartovm) [reply apps/chat-e2e/src/tests/chatSelectionFunctionality.test.ts]: done

**Comment by nartovm** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1884.nightly-test.deltixhub.io
>E2E tests status: failed

**Review by irinakartun — APPROVED** [review]

**Comment by nartovm** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1884.nightly-test.deltixhub.io
>E2E tests status: failed

**Pull request merged**